### DISCUSSION TITLE
OBS WHIP: Support for RTX?

### DISCUSSION THREAD

**Discussion by LVala** [obsproject/obs-studio#10568] [Development Questions] [answered]
Hi, I just wanted to make sure I understand how retransmissions work in OBS's WHIP implementation.
I've read (in the OBS WebRTC roadmap) that RTX is supported but "it doesn't use a distinct SSRC".

In the offer generated by OBS there's no "rtx" lines, so I assume that OBS will respond to NACKs by simply sending the exact same packet as the one that was lost (at least according the the sequence numbers from the NACK) once again? If so, isn't this a problem for SRTP and it's replay protection features (the "sliding window")?

I'm using OBS 30.1.2 on MacOS.

**Comment by pkviet** [comment]
@Sean-Der question for you

**Comment by Sean-Der** [comment, marked as answer]
Hi @LVala 

Everything you describe is correct! If the RTX falls outside the sliding window SRTP's replay protection will drop it.

In the future I would like to extend libdatachannel to use a distinct SSRC. If you are interested in doing that I would love your help!

**Reply by LVala** [reply]
Thanks for the clarification, I'll see what I can do!
